File: src/main/java/loxal/common/auth/AuthInfoSvcAsync.java
/*
 * Copyright 2012 Alexander Orlov <alexander.orlov@loxal.net>. All rights reserved.
 */

package loxal.common.auth;

import com.google.gwt.user.client.rpc.AsyncCallback;

/**
 * @author Alexander Orlov <alexander.orlov@loxal.net>
 */
public interface AuthInfoSvcAsync {
    /**
     * Google Authentication Service
     *
     * @param requestUri Request originator
     * @param callback   the callback to return Properties of the authenticated user
     */
    void getAuthInfo(String requestUri, AsyncCallback<AuthInfo> callback);
}

File: src/main/kotlin/loxal/common/auth/AuthInfoSvcAsync.kt
/*
 * Copyright 2015 Alexander Orlov <alexander.orlov@loxal.net>. All rights reserved.
 */

package loxal.common.auth

import com.google.gwt.user.client.rpc.AsyncCallback

public interface AuthInfoSvcAsync {
    /**
     * Google Authentication Service
     *
     * @param requestUri Request originator
     * @param callback   the callback to return Properties of the authenticated user
     */
    public fun getAuthInfo(requestUri: String, callback: AsyncCallback<AuthInfo>)
}